File: interview_cake/greedy_algorithms/fisher_yates.js
function getRandom(floor, ceiling) {
  return Math.floor(Math.random() * (ceiling - floor + 1)) + floor;
}

function shuffle(array) {

  // Shuffle the input in place
  let ceiling = array.length - 1;
  
  for (let i = 0; i < ceiling; i++) {
    let floor = i;
    random = getRandom(floor, ceiling)
    if (random != i) {
      [array[i], array[random]] = [array[random], array[i]];
    }
  }
}


const sample = [1, 2, 3, 4, 5];
console.log('Initial array: ', sample);
shuffle(sample);
console.log('Shuffled array: ', sample);

File: interview_cake/greedy_algorithms/fisher_yates.rb
def get_random(floor, ceiling)
  rand(floor..ceiling)
end

def shuffle(array)

  # Shuffle the input in place.
  ceiling = array.length - 1
  0.upto(array.length - 2) do |i|
    floor = i
    random = get_random(floor, ceiling)
    if random != i
      array[i], array[random] = array[random], array[i]
    end
  end

  array
end

sample_array = [1, 2, 3, 4, 5]
puts "Sample array: #{sample_array}"

puts 'Shuffling sample array...'
shuffle(sample_array)
puts sample_array.inspect
